prow/pod-utils/clone/clone.go

```diff
@@ -105,6 +112,16 @@ func gitCtxForRefs(refs prowapi.Refs, baseDir string, env []string) gitCtx {
 	if refs.CloneURI != "" {
 		g.repositoryURI = refs.CloneURI
 	}
+
+	if len(oauthToken) > 0 {
+		u, err := url.Parse(g.repositoryURI)
+		if err != nil {
+			logrus.WithError(err).Errorf("Cannot parse url from %s", g.repositoryURI)
+		} else {
+			g.repositoryURI = fmt.Sprintf("https://%s:x-oauth-basic@%s%s", oauthToken, u.Host, u.Path)
```

Comment:
Prefer to use [`UserPassword`](https://golang.org/pkg/net/url/#UserPassword)

Reply:
Interesting, iirc other places do `$botname:$token` and bend over backwards to obtain `$botname` even when it's not actually needed. TIL.